src/axes.jl

```diff
@@ -127,15 +127,17 @@ function get_guide(axis::Axis)
         axis[:unitformat] == :nounit 
         return axis[:guide]
     else
-        ustr = if Plots.backend_name() ≡ :pgfplotsx
+        unit = if axis[:unitformat] isa Function
+            unit = axis[:unit]
```

Comment:
One of these `unit = ` is superfluous. I would personally prefer keeping the second one and adding one at line 133 but it appears `= if` is pretty common in the rest of the code base so it's up to you :D